ql/src/java/org/apache/hadoop/hive/ql/security/authorization/plugin/sqlstd/SQLAuthorizationUtils.java

```diff
@@ -269,7 +269,19 @@ private static boolean isOwner(IMetaStoreClient metastoreClient, String userName
         thriftTableObj = metastoreClient.getTable(hivePrivObject.getDbname(),
             hivePrivObject.getObjectName());
       } catch (Exception e) {
-        throwGetObjErr(e, hivePrivObject);
+        boolean isTableExists = true;
+        try {
+          isTableExists = metastoreClient.tableExists(hivePrivObject.getDbname(), hivePrivObject.getObjectName());
+        } catch (TException ex){
+          throwGetObjErr(ex, hivePrivObject);
+        }
+        if(!isTableExists){
+          // Do not throw any exception when table object is not present.
+          // return true since the current user the owner of new table.
+          return true;
```

Comment:
I think instead of returning `true` here for non-existent tables; the "call-site" should check if the table already exists and make a decision based on that